src/jsonrpc-virtual-machine.cpp

```diff
@@ -171,6 +171,15 @@ jsonrpc_mg_mgr::jsonrpc_mg_mgr(std::string remote_address) {
 }
 
 jsonrpc_mg_mgr::~jsonrpc_mg_mgr() {
```

Comment:
Do you think this is a good idea? I am afraid somebody will use --max-mcycle and this commit will happen. Not sure this is a problem. But it some hidden behavior.

Reply:
Without this, we would to explicit do a `commit()` call in cartesi machine before exiting the script to kill the listening JSON-RPC socket. Also people who forgot this commit() call would have zombie forks that were not shutdown.

I could remove this, as long we document that forks could become zombies in case a commit/rollback is not executed, and we explicit call `commit()` to "close" a snapshot. Maybe I could even use to-be-closed variable on Lua to make sure a commit will be called, even in case of Lua errors.

Reply:
I would rather we not do this automatically. I think the right place for something like this is in cartesi-machine.lua. There, you can create an object to-be-closed with a metamethod that does this, after checking a command-line flag that lets the user disable the behavior.

Reply:
I removed this code, now the `cartesi-machine.lua` will clean this. I didn't feel the need to introduce a new command-line option to disable this behavior, meaning snapshots will be committed only when an advance state is accepted, all other cases will cause rollback in the remote machine before the cli exits, such as:
- Any raised lua error
- A machine that is stopped before an advance state is completed (due to max mcycle)
- Any inspects

However I took the opportunity to introduce a new option `--no-rollback`, to allow performing advance/inspect states on a local machine, with is a BIG disclaimer that state will never be reverted. This makes the cli a little more convenient to perform tests of advances/inspects without the need of spawning remote cartesi machines.